app/classifier.py

```diff
@@ -41,16 +67,43 @@ def classify(self, seq: str):
         }
 
         predictions = self.predictor(features)
-        results = predictions["probabilities"][0]
-        self.logger.debug(results)
+        probabilities = predictions["probabilities"][0]
+        self.logger.debug(probabilities)
 
         # map results back to topic strings, according to classifier metadata
-        out = list(zip((t.rstrip() for t in self.vocab), results))
-        out.sort(key=itemgetter(1), reverse=True)
+        # e.g. [('education', 0.8), ('rights of the child', 0.9)]
+        topic_probs = list(zip(
+            (topic.rstrip() for topic in self.vocab),
+            probabilities))
+        # sort the results in descending order of topic probability
+        topic_probs.sort(key=lambda tup: tup[1], reverse=True)
 
-        # TODO: filter results to those above a configurable threshold
-        out = list(o for o in out if o[1] > 0)
+        # discard all topic probability tuples who are too unlikely
+        topic_probs = [o for o in topic_probs if o[1] > RETAIN_THRESHOLD]
         self.logger.info(
             "Filtered %d results that were at or below the precision "
-            "threshold." % (len(self.vocab) - len(out)))
-        return out
+            "threshold." % (len(self.vocab) - len(topic_probs)))
+        return topic_probs
+
+
+@classify_bp.route('/classify', methods=['POST'])
+def classify():
+    # request.get_json: {"seq"="hello world"}
+    error = None
+    data = request.get_json()
+
+    with open(app.config["MODEL_CONFIG_PATH"]) as f:
```

Comment:
should the MODEL_CONFIG_PATH be part of the request? Or at least overridable?

Reply:
done